lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3918,75 +3915,18 @@ void AttributeChecker::visitDerivativeAttr(DerivativeAttr *attr) {
     return;
   }
 
-  // Try to find a `@differentiable` attribute on the original function with the
-  // same differentiation parameters.
-  DifferentiableAttr *da = nullptr;
-  for (auto *cda : originalAFD->getAttrs().getAttributes<DifferentiableAttr>())
-    if (checkedWrtParamIndices == cda->getParameterIndices())
-      da = const_cast<DifferentiableAttr *>(cda);
-  // If the original function does not have a `@differentiable` attribute with
-  // the same differentiation parameters, create one.
-  // TODO(TF-835): Lower `@derivative` attributes directly to SIL
-  // differentiability witnesses during SILGen instead of generating implicit
-  // `@differentiable` attributes.
-  if (!da) {
-    da = DifferentiableAttr::create(
-        originalAFD, /*implicit*/ true, attr->AtLoc, attr->getRange(),
-        /*linear*/ false, checkedWrtParamIndices, /*jvp*/ None,
-        /*vjp*/ None, derivative->getGenericSignature());
-    switch (kind) {
-    case AutoDiffDerivativeFunctionKind::JVP:
-      da->setJVPFunction(derivative);
-      break;
-    case AutoDiffDerivativeFunctionKind::VJP:
-      da->setVJPFunction(derivative);
-      break;
-    }
-    auto insertion = Ctx.DifferentiableAttrs.try_emplace(
-        {originalAFD, checkedWrtParamIndices}, da);
-    // Valid `@differentiable` attributes are uniqued by their parameter
-    // indices. Reject duplicate attributes for the same decl and parameter
-    // indices pair.
-    if (!insertion.second && insertion.first->getSecond() != da) {
-      diagnoseAndRemoveAttr(da, diag::differentiable_attr_duplicate);
-      diagnose(insertion.first->getSecond()->getLocation(),
-               diag::differentiable_attr_duplicate_note);
-      return;
-    }
-    originalAFD->getAttrs().add(da);
+  // Valid `@derivative` attributes are uniqued by original function and
+  // parameter indices. Reject duplicate attributes.
```

Comment:
This comment is confusing because it doesn't mention `kind` which is also in the uniquification key. I suggest just saying "Reject duplicate attributes" because that will never go stale even if we change the uniquification key in the future. The person reading the comment can easily look at the key in the code to see what it's uniqued by.

Reply:
Good catch! Done in https://github.com/apple/swift/pull/28621/commits/f9ced25561c3bbb41ca628a9e41f72f13137562c.